iexec/src/main/scala/gridscale/iexec/package.scala

```diff
@@ -0,0 +1,107 @@
+package gridscale
+
+import effectaside._
+import gridscale.cluster.BatchScheduler.BatchJob
+import gridscale.cluster.{ BatchScheduler, HeadNode }
+import gridscale.tools._
+import monocle.macros._
+import squants._
+import squants.information._
+
+package object iexec {
+
+  case class IEXECJobDescription(
+    workDirectory: String,
+    IexecFilesPath: String,
+    dappAddress: String,
+    arguments: String,
+    dappCost: Int)
+
+  object impl {
+
+    def toIexec[S](server: S, description: IEXECJobDescription)(implicit hn: HeadNode[S], system: Effect[System]) = {
+
+      loginIexecAccount(server, description)
+      verifyIexecAccountBalance(server, description)
+
+      s"""
+         |#!/bin/bash
+         |export PATH="${description.IexecFilesPath}:$${PATH}"
+         |cd ${description.workDirectory}
+         |DEBUG='iexec:submit' iexec submit ${description.arguments} --dapp ${description.dappAddress}
+       """.stripMargin
+    }
+
+    def loginIexecAccount[S](server: S, description: IEXECJobDescription)(implicit hn: HeadNode[S], system: Effect[System]) = {
+      if (!new java.io.File(s"""/${description.workDirectory}/account.json""").exists) {
+        hn.execute(server, s"""cd ${description.workDirectory} && export PATH="${description.IexecFilesPath}:$${PATH}" && iexec account login""")
+      }
+    }
+
+    def populateIexecAccount[S](server: S, description: IEXECJobDescription, amount: Int)(implicit hn: HeadNode[S], system: Effect[System]) = {
+      val cmdRet = hn.execute(server, s"""cd ${description.workDirectory} && export PATH="${description.IexecFilesPath}:$${PATH}" && iexec account allow ${amount}""")
+    }
+
+    def verifyIexecAccountBalance[S](server: S, description: IEXECJobDescription)(implicit hn: HeadNode[S], system: Effect[System]) = {
+      val ExecutionResult(ret, out, error) = hn.execute(server, s"""cd ${description.workDirectory} && export PATH="${description.IexecFilesPath}:$${PATH}" && iexec account show""")
+
+      var ropstenNetworkIndex = 2
+
+      val ropstenBalanceLine = out.split("\n")(ropstenNetworkIndex)
+      val rlcBalance = ropstenBalanceLine.substring(ropstenBalanceLine.indexOf(":")+1, ropstenBalanceLine.indexOf(" nRLC")).trim.toInt
+
+      assert(rlcBalance >= description.dappCost, s"iExec account does not have enough tokens to execute this DApp, account needs at least ${description.dappCost} but has ${rlcBalance}")
+    }
```

Comment:
Perfect! Now you just need another function to set the allowance. It would not be called automatically in submit but available to be called manually by the user.